### PR TITLE
Fix 404s and redirecting links

### PR DESCRIPTION
**Pull request by rmoff**
Fix the links identified by lychee here https://github.com/rmoff/lakeFS/actions/runs/4564037405#summary-12393998465

Some are genuinely 404, others are logically broken and rely on jekyll redirects.

### CODE REVIEW

File: docs/reference/authentication.md [hunk 1]
```diff
@@ -137,12 +137,12 @@ OIDC provider. When a user first logs in through the provider, a corresponding u
 
 #### Notes
 {: .no_toc}
-1. As always, you may choose to provide these configurations using [environment variables](./configuration.md#using-environment-variables).
+1. As always, you may choose to provide these configurations using [environment variables](./configuration.html).
```
Comment (arielshaqed) [inline docs/reference/authentication.md:140]: AFAICS this loses the anchor.

**Review by arielshaqed — APPROVED** [review]
Nice!

I'm not sure about all the extension hacking: if it works without this, why do we want to fix it?  Ideally we would have the simplest possible form to write, and then have Jekyll fix it up for us when generating the website.

Assuming we _do_ want to fix: If we start using `.html` extensions then I worry that people will _still_ get it wrong and keep on adding the easier `.md` extensions (or no extension at all, if that works).  If these are truly undesirable, let's open an issue to scan for them so that humans cannot repeat this mistake.

File: docs/reference/authentication.md (hunk 1 above)
Comment (rmoff) [reply docs/reference/authentication.md:140]: The target no longer exists :)

**Comment by rmoff** [comment]
Thanks @arielshaqed. Tracking the point about extensions here: https://github.com/treeverse/lakeFS/issues/5614

**Pull request merged**